docs/blog/pydantic-v2-alpha.md

```diff
@@ -0,0 +1,182 @@
+* **JSON Schema** - generation of JSON Schema is much improved and ready to test.
+* **Generic Models** - are much improved and ready to test.
+* **Recursive Models** - and validation of recursive data structures is much improved and ready to test.
+* **Custom Types** - custom types have a new interface and are ready to test.
+* **Custom Field Modifiers** - used via `Annotated[]` are working and in use in Pydantic itself. 
+* **Validation without a BaseModel** - the new `AnalyzedType` class allows validation without the need for a `BaseModel` class, and it's ready to test.
+* **TypedDict** - we now have full support for `TypedDict` via `AnalyzedType`, it's ready to test.
+
+### Still under construction
+
+* **Documentation** - we're working hard on full documentation for V2, but it's not ready yet.
+* **Conversion Table** - a big addition to the documentation will be a conversion table showing how types are coerced, this is a WIP.
+* **BaseSettings** - `BaseSettings` will move to a separate `pydantic-settings` package, it's not yet ready to test.
+* **validate_arguments** - the `validate_arguments` decorator remains and is working, but hasn't been updated yet.
+* **Hypothesis Plugin** - the Hypothesis plugin is yet to be updated.
+* **computed fields** - we know a lot of people are waiting for this, we will include it in Pydantic V2.
+* **Error messages** - could use some love, and links to docs in error messages are still to be added.
+* **Migration Guide** - we have some pointers below, but this needs completing.
+
+## Migration Guide
+
+**Please note:** this is just the beginning of a migration guide. We'll work hard up to the final release to prepare
+a full migration guide, but for now the following pointers should be some help while experimenting with V2.
+
+### Changes to BaseModel
+
+* Various method names have been changed; `BaseModel` methods all start with `model_` now. 
+  Where possible, we have retained the old method names to help ease migration, but calling them will result in `DeprecationWarning`s.
+  * Some of the built-in data loading functionality has been slated for removal. 
+    In particular, `parse_raw` and `parse_file` are now deprecated. You should load the data and then pass it to `model_validate`.
+* The `from_orm` method has been removed; you can now just use `model_validate` (equivalent to `parse_obj` from Pydantic V1) to achieve something similar, 
+  as long as you've set `from_attributes=True` in the model config.
+* The `__eq__` method has changed for models; models are no longer considered equal to the dicts.
+* Custom `__init__` overrides won't be called. This should be replaced with a `@root_validator`.
+* Support for `__root__` models has been removed. However, you can effectively achieve something equivalent through the use of 
+  `@root_validator`, `@model_serializer`, and `__pydantic_modify_json_schema__`. More details will be available soon.
```

Comment:
@samuelcolvin any thoughts on how to provide these extra details? (This is basically ready now, but not quite sure how to explain; doesn't seem appropriate to include the full set of functions to define in-line here..)

Reply:
I'm tentatively planning to add a permalink to the implementation in one of the tests that uses a root model (after #5340 is merged), rather than an issue/discussion, and wait for someone to ask if they have questions/issues. But I could do that instead.

Reply:
See https://github.com/pydantic/pydantic/pull/5357 for an update to this

Reply:
#5357 lgtm.